bin/poeditor-pull.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash                                                                                                                                                                                     
+CURL='/usr/bin/curl'
+POEDITOR_EXPORT_URL="https://api.poeditor.com/v2/projects/export"
+API_TOKEN="$POEDITOR_API_TOKEN" # may be set as environment variable
+OUTPUT_DIR="." # default before argument processing
+
+# argument processing from https://stackoverflow.com/a/14203146
+while [[ $# -gt 1 ]]
+do
+key="$1"
+
+case $key in
+    -a|--api_token)
+    API_TOKEN="$2"
+    shift # past argument
+    ;;
+    -p|--project_id)
+    PROJECT_ID="$2"
+    shift # past argument
+    ;;
+    -l|--language)
+    LANGUAGE="$2"
+    shift # past argument
+    ;;
+    -o|--output_dir)
+    OUTPUT_DIR="$2"
+    shift # past argument
+    ;;
+esac
+shift # past argument or value
+done
+
+CURLARGS="-X POST -d id=$PROJECT_ID -d language=$LANGUAGE -d type=key_value_json -d api_token=$API_TOKEN"
+# echo "CURLARGS='$CURLARGS'"
+
+# retrieve the initial API response
+RESPONSE=$($CURL $CURLARGS $POEDITOR_EXPORT_URL)
```

Comment:
You can add the same check as above to each of the next 3 lines

`[[ -z "$VARIABLE_NAME" ]] && { echo "VARIABLE_NAME is empty" ; exit 1; }`